src/Polymorphism.hs

```diff
@@ -1,24 +1,30 @@
-module Polymorphism where
+module Polymorphism
+  ( nameOfPolymorphicFunction,
+  )
+where
 
-import Lookup
+import Env as E
 import Obj
 import Types
 
 -- | Calculate the full, mangled name of a concretized polymorphic function.
--- | For example, The 'id' in "(id 3)" will become 'id__int'.
--- | This function uses the 'multiLookupALL' function which gives it acces to
--- | modules that are not imported. This allows it to access 'delete' functions
--- | and similar for internal use.
-
--- | TODO: Environments are passed in different order here!!!
+-- For example, The 'id' in "(id 3)" will become 'id__int'.
+--
+-- This function uses findPoly, which gives it access to *all* possible
+-- environments in the given input environment (children, (modules) parents,
+-- and use modules). This allows it to derive the correct name for functions
+-- that may be defined in a different environment.
+--
+-- TODO: Environments are passed in different order here!!!
 nameOfPolymorphicFunction :: TypeEnv -> Env -> Ty -> String -> Maybe SymPath
 nameOfPolymorphicFunction _ env functionType functionName =
-  let foundBinders = multiLookupEverywhere functionName env
-   in case filter ((\(Just t') -> areUnifiable functionType t') . xobjTy . binderXObj . snd) foundBinders of
-        [] -> Nothing
-        [(_, Binder _ (XObj (Lst (XObj (External (Just name)) _ _ : _)) _ _))] ->
+  let foundBinder =
+        (E.findPoly env functionName functionType)
```

Comment:
I see this "pattern" pop up in a bunch of places – does this mean that we do `findPoly` first in the env and then in the global env (progenitor) if nothing was found in env?

Reply:
Excellent!